---
title: "Proteomics"
author: "YJ"
date: "30 5 2022"
output: html_document
editor_options: 
  chunk_output_type: console
---

# New proteomic analysis
# Using package DEP
# The difference is that the NA value is filtered and the data set was normalized
# link for proteomic analysis https://bioconductor.org/packages/devel/bioc/vignettes/DEP/inst/doc/DEP.html

```{r,include=F}
library(readxl)
library(DEP)
library(tidyverse)
library(ggplot2)
library(ggrepel)
library(pheatmap)
library(reshape2)
library(ggfortify)
library(broom)

```


```{r}
#read raw file
library(readr)
proteinGroups <- read_delim("./raw/Proteomics/raw/proteinGroups.txt", 
                            delim = "\t", escape_double = FALSE, 
                            col_types = cols(`Only identified by site` = col_character(),Reverse=col_character(),`Potential contaminant` =col_character()), 
                            trim_ws = TRUE)
Protein_ID <- read_delim("./raw/Proteomics/raw/Protein_ID.txt", 
     delim = "\t", escape_double = FALSE, 
     col_types = cols(protein_ID = col_character()), 
     trim_ws = TRUE)
colnames(proteinGroups)
DB_meta <- read_excel("./raw/720020 AstUP results_NMBU_NTNU.xlsx",sheet = 1)

# First is to annotate uniprot with gene name
# This was done by uniprot website, transfer uniprot to gene name
Uniprot_to_genename <- read.table("../../Proteomics/raw/Unipro_to_GeneName.txt",header = T)
Data <- proteinGroups%>%
  mutate(Uniprot=gsub(";.*","",`Protein IDs`))

Data <- Data%>%
  filter(is.na(Reverse))%>%
  filter(is.na(`Potential contaminant`))

Data<- Data%>%
  merge(Uniprot_to_genename,by="Uniprot")



# remove sample #24 becasue low protein identified
Data<- Data%>%
  dplyr::select(-`LFQ intensity 24_20210308111738`)

colnames_new <- colnames(Data%>%
           dplyr::select(contains("LFQ intensity")))%>%
  as.data.frame()%>%
  mutate(protein_ID=gsub("LFQ intensity ","",.),
         protein_ID=gsub("_.*","",protein_ID))%>%
  rename("old_col"=".")%>%
  left_join(Protein_ID,by="protein_ID")%>%
  separate(sample_ID, into=c("Tank","other","Fish"),sep="_")%>%
  mutate(Tank=paste("Tank",Tank,sep = ""))%>%
  left_join(DB_meta%>%
              dplyr::select(Tank,Treatment),by="Tank")%>%
  mutate(Treatment=factor(Treatment,levels = c("Control","Phytosterol","DHA","Cholesterol")))%>%
  arrange(Treatment)%>%
  unique()%>%
  mutate(replicate=c(1:8,1:9,1:9,1:9),
         new_col = paste(Treatment,replicate,sep = "_"))%>%
  dplyr::select(old_col,new_col)

Data <- Data%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("old_col")%>%
  left_join(colnames_new,by="old_col")%>%
  mutate(new_col=ifelse(is.na(new_col),old_col,new_col))%>%
  column_to_rownames("new_col")%>%
  dplyr::select(-old_col)%>%
  t()%>%
  as.data.frame()
#keep column needed
Data <- Data%>%
  dplyr::select(Uniprot,Genename,`Fasta headers`, contains("Control"),contains("Phytosterol"),contains("DHA"),contains("Cholesterol"))%>%
  dplyr::select(-Cholesterol_9)


# Remove duplication
Data$Genename %>% duplicated() %>% any()
Data %>% group_by(Genename) %>% summarize(frequency = n()) %>% 
  arrange(desc(frequency)) %>% filter(frequency > 1)
data_unique <- make_unique(Data, "Genename", "Uniprot", delim = ";")

# Generate a SummarizedExperiment object
  # Getting all sample column
Diet_columns <- grep("Control|Phytosterol|DHA|Cholesterol",colnames(data_unique))
  
  # generate design matrix
  Design_matrix <- colnames(data_unique[,Diet_columns])%>%
    as.data.frame()%>%
    rename("label"=".")%>%
    mutate(condition = gsub("_.*","",label)%>%
             factor(levels = c("Control","Phytosterol","DHA","Cholesterol")),
           replicate = gsub(".*_","",label)%>%
             as.numeric())%>%
    arrange(condition)

#lets try with only cholesterol
Diet_columns<-which(colnames(data_unique)%in%Design_matrix$label)

#make SummarizedExperiment
data_unique[Diet_columns] <- sapply(data_unique[Diet_columns],as.numeric)
data_unique[data_unique==0] <- NA
data_se <- make_se(data_unique, Diet_columns, Design_matrix)

#Filter on missing values
#The dataset contains proteins which are not quantified in all replicates. Some proteins are even only quantified in a single replicate.
# Plot a barplot of the protein identification overlap between samples
plot_frequency(data_se)

# Filter for proteins that are quantified in at least 2/3 of the samples.
data_filt <- filter_proteins(data_se, "fraction", min = 0.5)



# 2376 protein were identified
# Plot a barplot of the number of identified proteins per samples
plot_numbers(data_filt)

# Plot a barplot of the protein identification overlap between samples
plot_coverage(data_filt)

#The data is background corrected and normalized by variance stabilizing transformation (vsn).
# Normalize the data
data_norm <- normalize_vsn(data_filt)
meanSdPlot(data_norm)
plot_normalization(data_filt, data_norm)

# Plot a heatmap of proteins with missing values
plot_missval(data_norm)

# Plot intensity distributions and cumulative fraction of proteins with and without missing values
plot_detect(data_filt)

# Impute missing data using the k-nearest neighbour approach (for MAR)
knn_imputation <- impute(data_filt, fun = "knn", rowmax = 0.9)

#The effect of data imputation on the distributions can be visualized.
# Plot intensity distributions before and after imputation
plot_imputation(data_norm,knn_imputation)

# Differential enrichment analysis  based on linear models and empherical Bayes statistics
# Test every sample versus control
data_diff <- test_diff(knn_imputation, type = "control", control = "Control")


# Denote significant proteins based on user defined cutoffs
dep <- add_rejections(data_diff,alpha = 0.05)

plot_pca(dep, x = 1, y = 2, n = 100, point_size = 4,indicate = c("condition"),label = T)

#plot single protein
plot_single(dep, proteins = "acat1", type = "centered")

# Plot the first and second principal components

data_results <- get_results(dep)%>%
  left_join(Data%>%
              mutate(ID=Uniprot)%>%
              dplyr::select(ID,`Fasta headers`))%>%
  mutate(Genename=gsub("OS=.*","",`Fasta headers`),
         Genename=gsub(".*SALSA |.*ONCMY ","",Genename)
         )%>%
  mutate(Cholesterol_vs_Control_p.adj=p.adjust(Cholesterol_vs_Control_p.val,method = "BH"),
        Phytosterol_vs_Control_p.adj=p.adjust(Phytosterol_vs_Control_p.val,method = "BH"),
        DHA_vs_Control_p.adj=p.adjust(DHA_vs_Control_p.val,method = "BH"))

plot_cor(dep, significant = TRUE, lower = 0, upper = 1, pal = "Reds")

plot_volcano(dep, contrast = "Cholesterol_vs_Control", label_size = 2, add_names = TRUE)

plot_heatmap(dep)

DEP_Cho_table <- data_results%>%
  dplyr::select(name,ID,Genename,Cholesterol_vs_Control_p.val,Cholesterol_vs_Control_p.adj,Cholesterol_vs_Control_ratio)%>%
  filter(Cholesterol_vs_Control_p.val<0.001)

Count_table <- dep@assays@data@listData[[1]]%>%
  as.data.frame()%>%
  rownames_to_column("name")%>%
  filter(name%in%DEP_Cho_table$name)%>%
  left_join(DEP_Cho_table%>%
              dplyr::select(name,Genename),by="name")%>%
  dplyr::select(-name)%>%
  dplyr::select(contains("Control"),contains("Cholesterol"),Genename)
pheatmap(Count_table[,1:16],cluster_cols = F,scale = "row",labels_row = Count_table$Genename,color=colorRampPalette(c("#018571", "#f5f5f5", "#a6611a"))(50))

DEP_PS_table <- data_results%>%
  dplyr::select(name,ID,Genename,Phytosterol_vs_Control_p.val,Phytosterol_vs_Control_ratio)%>%
  filter(Phytosterol_vs_Control_p.val<0.001)

Count_table <- dep@assays@data@listData[[1]]%>%
  as.data.frame()%>%
  rownames_to_column("name")%>%
  filter(name%in%DEP_Cho_table$name)%>%
  left_join(DEP_Cho_table%>%
              dplyr::select(name,Genename),by="name")%>%
  dplyr::select(-name)%>%
  dplyr::select(contains("Control"),contains("Cholesterol"),Genename)
pheatmap(Count_table[,1:16],cluster_cols = F,scale = "row",labels_row = Count_table$Genename,color=colorRampPalette(c("#018571", "#f5f5f5", "#a6611a"))(50))

# Heatmap based on normalized data (missing data not imputed)
# test linear regression between plasma levels and differential proteins in intestine
# Here we tried to identify protein that response to plasma astaxanthin levels regardless of dietary treatment

Data_TB <- data_norm@assays@data@listData[[1]]%>%
  as.data.frame()%>%
  rownames_to_column("name")%>%
  melt(id.vars=c("name"))%>%
  rename("Protein_Norm"="value",
         "new_col"="variable")%>%
  left_join(colnames_new,by="new_col")%>%
    mutate(protein_ID=gsub("LFQ intensity ","",old_col),
         protein_ID=gsub("_.*","",protein_ID))%>%
  left_join(Protein_ID,by="protein_ID")%>%
  mutate(Tank=gsub("_.*","",sample_ID),
         Tank=paste("Tank",Tank,sep = ""),
         Fish=gsub(".*_","",sample_ID))%>%
  left_join(DB_meta%>%
                  mutate(Fish= as.character(Fish)),
           by=c("Tank","Fish"))%>%
  filter(!is.na(Plasma_Asta)&!is.na(Protein_Norm))

# Do heatmap for all proteins
Data_TB_Heat <- Data_TB%>%
  dplyr::select(Tank,Fish,Plasma_Asta,Treatment,Protein_Norm,name)%>%
  dcast(Fish+Tank+Treatment+Plasma_Asta~name,value.var = "Protein_Norm")%>%
  arrange(Plasma_Asta)%>%
  select_if(~ !any(is.na(.)))

pheatmap(Data_TB_Heat[,5:1119],cluster_rows = F,show_colnames = F,labels_row = Data_TB_Heat$Treatment,scale = "column")

# Do correlation for each protein
Data_reg <- Data_TB %>% 
  nest_by(name) %>%
  mutate(mod = list(lm(Protein_Norm ~ Plasma_Asta+Treatment, data = data))) %>%
  summarize(tidy(mod))%>%
  filter(term=="Plasma_Asta")

Data_reg_sort <- Data_reg%>%
  left_join(Data%>%
              mutate(name=Genename,
                Genename=gsub("OS=.*","",`Fasta headers`),
                Genename=gsub(".*SALSA |.*ONCMY ","",Genename))%>%
              dplyr::select(name,Genename),by="name")%>%
  arrange(p.value)%>%
  as.data.frame()%>%
  mutate(fdr= p.adjust(p.value,method = "BH"))%>%
  slice(1:100)

Data_reg_sort$fdr <- p.adjust(Data_reg_sort$p.value,method = "BH")

Data_TB_Heat_sort <- Data_TB_Heat[,colnames(Data_TB_Heat)%in%Data_reg_sort$name]%>%
  cbind(Data_TB_Heat%>%dplyr::select(Plasma_Asta))

library(ComplexHeatmap)
heatmap(Data_TB_Heat[,"Plasma_Asta"]%>%as.matrix())

pheatmap(Data_TB_Heat_sort[,1:60],cluster_rows = F,scale = "column",color=colorRampPalette(c("navy", "white", "red"))(50))


# Single protein 
SinglePro<- Data_TB%>%
  filter(name=="LOC106591134")

ggplot(SinglePro,aes(x=Plasma_Asta,y=Protein_Norm))+
  geom_point()+
  geom_smooth(formula = y~x,method='lm')

ggplot(SinglePro,aes(x=Treatment,y=Protein_Norm,fill=Treatment))+
  geom_boxplot()+
  geom_point()


# plot single protein
# VTG
vtg <- Data%>%
  filter(Genename=="LOC110508649")%>%
  melt(id.vars=c("Uniprot","Genename","Fasta headers"))%>%
  mutate(Diet=gsub("_.*","",variable),
         value=as.numeric(value))%>%
  arrange(value)

ggplot(vtg,aes(x=variable,y=value,color=Diet))+
  geom_point()+
  coord_flip()+
  geom_text_repel(aes(label=variable))+
  ggtitle("vtg3 protein expression 2")


Data_corr <- Count_table_scaled%>%
  rownames_to_column("name")%>%
  melt(id.vars=c("name"))%>%
  rename("Protein_scaled"="value")%>%
  separate(variable,into=c("condition","replicate"),sep="_")%>%
  left_join(Design_matrix%>%
              mutate(replicate=as.character(replicate)),by=c("condition","replicate"))%>%
    mutate(protein_ID=gsub("LFQ intensity ","",label),
         protein_ID=gsub("_.*","",protein_ID))%>%
  left_join(Protein_ID,by="protein_ID")%>%
  mutate(Tank=gsub("_.*","",sample_ID),
         Tank=paste("Tank",Tank,sep = ""),
         Fish=gsub(".*_","",sample_ID))%>%
  left_join(meta_new_scaled,by=c("Tank","Fish"))%>%
  filter(!is.na(Plasma_asta))

# Do correlation for each protein
Data_reg <- Data_corr %>% 
  nest_by(name) %>%
  mutate(mod = list(lm(Protein_scaled ~ Plasma_asta+Treatment, data = data))) %>% 
  summarize(tidy(mod))%>%
  filter(term=="Plasma_asta")%>%
  left_join(data_results,by="name")

# Single protein 
SinglePro<- Count_table%>%
  rownames_to_column("name")%>%
  filter(name=="acat1")%>%
  melt(id.vars=c("name"))%>%
  rename("protein"="value")%>%
  separate(variable,into=c("condition","replicate"),sep="_")%>%
  left_join(Design_matrix%>%
              mutate(replicate=as.character(replicate)),by=c("condition","replicate"))%>%
    mutate(protein_ID=gsub("LFQ intensity ","",label),
         protein_ID=gsub("_.*","",protein_ID))%>%
  left_join(Protein_ID,by="protein_ID")%>%
  mutate(Tank=gsub("_.*","",sample_ID),
         Tank=paste("Tank",Tank,sep = ""),
         Fish=gsub(".*_","",sample_ID))%>%
  merge(DB_meta%>%
              melt(measure.vars=c("Asta_plasma_fish1","Asta_plasma_fish2"))%>%
              mutate(Fish=gsub(".*fish","",variable)),by=c("Tank","Fish"))%>%
  rename("Plasma_asta"="value")%>%
  mutate(Protein_scaled=scale(protein),
         asta_scaled=scale(Plasma_asta))

ggplot(SinglePro,aes(x=asta_scaled,y=Protein_scaled))+
  geom_point(aes(color=Treatment))+
  geom_smooth(formula = y~x)


ggplot(SinglePro,aes(x=Treatment,y=Protein_scaled))+
  geom_boxplot()+
  geom_point()



count_p <- Count_table%>%
  filter(Cholesterol_vs_Control_p.val<0.02)%>%
  dplyr::select(Genename,Control_1,Control_2,Control_3,Control_4,Control_5,Control_6,Control_7,Control_8,Cholesterol_1,Cholesterol_2,Cholesterol_3,Cholesterol_4,Cholesterol_5,Cholesterol_6,Cholesterol_7,Cholesterol_8,Cholesterol_9)

pheatmap(count_p[2:18],labels_row = count_p$Genename,cluster_cols = F,gaps_col  = c(8),scale = "row", color = colorRampPalette(c("navy", "white", "firebrick3"))(50))



# PCA of normalized data
pca_table <- proteinGroups%>%
  dplyr::select(contains("LFQ"))
ntop <- 200
rv <- apply(pca_table, 1, var)
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))]
mat <- t( pca_table[select, ] )
pca_all <- prcomp( mat, center = TRUE )

pca <-prcomp(mat, center = TRUE )$x[,1:6] %>% 
  as.data.frame()%>% 
  rownames_to_column('sample') %>%
  mutate(protein_ID=gsub("LFQ intensity ","",sample),
         protein_ID=gsub("_.*","",protein_ID))%>%
  left_join(Protein_ID,by="protein_ID")%>%
  mutate(Tank=gsub("_.*","",sample_ID),
         Tank=paste("Tank",Tank,sep = ""))%>%
  left_join(DB_meta,by="Tank")%>%
  dplyr::mutate(percPC1 = (pca_all$sdev[1]^2/sum(pca_all$sdev^2)),
                percPC2 = (pca_all$sdev[2]^2/sum(pca_all$sdev^2)))

ggplot(pca, aes(PC1, PC2)) + 
  geom_hline(yintercept=0, colour="gray75") + 
  geom_vline(xintercept=0, colour="gray75") + 
  geom_point(aes(color=Treatment),size=4) + 
  coord_equal()+
  scale_fill_manual(values=c("#bababa","#fdae61","#b8e186","#a6cee3"))+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"))





```





