---
title: "Redtrout_RNAseq"
author: "YJ"
date: "10 3 2021"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ggplot2)
library(edgeR)
library(tidyverse)
library(readxl)
library(rentrez)
library(reshape2)
library(multcomp)
library(DescTools)
library(pheatmap)

```


## General data
```{r}
# Read meta table
## WB means whole body; color means total pigment, asta means astaxanthin

# One-way anova was used 
# Dunnett’s was used as post-hoc test where all treatments were compared to the control

DB_meta <- read_excel("../meta/720020 AstUP results_NMBU_NTNU.xlsx",sheet = 1)

DB<- DB_meta%>%
  dplyr::select(-Plasma_Asta,-Fish)%>%
  melt(id.vars=c(1:2))%>%
  filter(!is.na(value))%>%
  mutate(variable=gsub("_fish1","",variable),
         variable=gsub("_fish2","",variable))%>%
  group_by(Treatment,variable)%>%
  mutate(Mean=mean(value),
         SD=sd(value),
         Group1=gsub("_.*","",variable),
         Group2=gsub(".*_","",variable))%>%
  ungroup()%>%
  filter(!Group1=="Diet"&!Group2=="Yttrium")%>%
  mutate(variable=factor(variable),
         Treatment=factor(Treatment, levels = c("Control",
                                                "Cholesterol",
                                                "Phytosterol",
                                                "DHA")))
# Anova and post-hoc
# One way anova was used for all samples
# Dunnett’s correction was used as post-hoc test, usiing control diet as a reference

aov_DB<- DB%>%
    dplyr::select(Treatment,variable, value)%>%
    group_by(variable)%>%
  do(aov = broom::tidy(aov(value ~ Treatment, data = .))) %>%
  unnest(aov) %>% 
  filter(!is.na(statistic)) %>% 
  dplyr::select(-df, -sumsq, -meansq, -statistic) %>% 
  mutate(term = paste0("lm_", gsub(":", "_", term))) %>% 
  spread(term, p.value) 

post<- list()
aov<- list()
post_DB <- DB%>%
    dplyr::select(Treatment,variable, value)%>%
  split(list(.$variable))

for (i in 1:length(post_DB)){
  aov[[i]]<- aov(value~Treatment,data = post_DB[[i]])
  post[[i]]<- TukeyHSD(aov[[i]])[[1]][,"p adj"]%>%
    t()%>%
    as.data.frame()%>%
    mutate(variable=names(post_DB)[[i]])
}

post_all <- do.call(rbind.data.frame,post)
  
DB_sum <- DB%>%
  dplyr::select(-Tank,-value)%>%
  unique()%>%
  mutate(value=paste(round(Mean, digits=1),"±",round(SD, digits=1)))%>%
  dplyr::select(Treatment, variable,value)%>%
  left_join(aov_DB,by=c("variable"))%>%
  dcast(variable+lm_Treatment~Treatment)%>%
  left_join(post_all,by="variable")%>%
  mutate(lm_p = format(round(lm_Treatment,digits=2),scientific = T),
         CH_Con_p=format(round(`Cholesterol-Control`,digits=2),scientific = T),
         PS_Con_p=format(round(`Phytosterol-Control`,digits=2),scientific = T),
         DHA_Con_p=format(round(`DHA-Control`,digits=2),scientific = T))
colnames(DB_sum)<- c("Measurement","")



#Astaxanthin figures
DB_select <- DB%>%
  filter(variable=="Asta_plasma"|
           variable=="Color (mg/kg)"|
           variable=="Feces_Asta"|
           variable=="WB_Asta"|
           variable=="Asta_retention")

DB_sum <-DB_select %>%
  dplyr::select(Mean,SD,variable,Treatment)%>%
  unique()%>%
  group_by(variable)%>%
  mutate(sum=Mean+SD,
         min=0,
         max=max(sum)*1.2)%>%
  ungroup()


ggplot(DB_sum, aes(x=Treatment, y=Mean,fill=Treatment))+
  geom_segment(aes(x=Treatment, xend=Treatment, y=min, yend=max), colour="white")+
  #geom_boxplot()+
  geom_errorbar(aes(ymax=Mean+SD,ymin=Mean),width=0.4)+
  geom_bar(stat = "identity",width = 0.8)+
  #geom_point(data = DB_select,aes(x=Treatment, y=value))+
  theme_bw()+
  theme(strip.text.x = element_text(size = 12, color = "black", face = "italic"),
        strip.background = element_blank(),
        text = element_text(color = "black",size=12),
        axis.text = element_text(color = "black",size=12))+
  scale_x_discrete(labels=c("Con","Chole","Phyto","DHA"))+
  facet_wrap(~variable,scales = "free_y",ncol = 6)+
  scale_fill_manual(values=c("#bababa","#fdae61","#b8e186","#a6cee3"))+
  scale_y_continuous(expand = c(0,0))


## Salmon Fan
salmofan <- read_excel("../meta/720020 AstUP results_NMBU_NTNU.xlsx",sheet = 2)%>%
  mutate(Treatment=factor(Treatment, levels = c("Control",
                                                "Cholesterol",
                                                "Phytosterol",
                                                "DHA")))%>%
  rename("value"="Salmofan")%>%
  group_by(Treatment)%>%
  mutate(Mean=mean(value),
         SD=sd(value))%>%
  ungroup()

salmofan_aov <- aov(value~Treatment,salmofan)

salmofan_posthoc <- DunnettTest(value~Treatment,data=salmofan, control="Control")[[1]]

ggplot(data = salmofan,aes(x=Treatment, y=value, fill=Treatment))+
  geom_violin(aes(x=Treatment,y=value))+
  #geom_boxplot()+
  #geom_errorbar(aes(ymax=Mean+SD,ymin=Mean-SD),width=0.2)+
  #geom_point(aes(x=Treatment, y=Mean))+
  #geom_bar(stat = "summary",fun="mean")+
  geom_dotplot(aes(x=Treatment,y=value),fill="black",binaxis = "y",
               stackdir = "center",
               dotsize = 0.5,binwidth = 0.3)+
  theme_bw()+
  scale_x_discrete(labels=c("Con","Chole","Phyto","DHA"))+
  scale_y_continuous(limits = c(19,32),expand = c(0,0))+
  ylab("Salmofan")+
  xlab("")



## Minolta
Minolta <- read_excel("../meta/720020 AstUP results_NMBU_NTNU.xlsx",sheet = 3)%>%
  mutate(Treatment=factor(Treatment, levels = c("Control",
                                                "Cholesterol",
                                                "Phytosterol",
                                                "DHA")))%>%
  melt(id.vars=c("Treatment","Tank","Position"))%>%
  group_by(Treatment, Position, variable)%>%
  mutate(Mean=mean(value),
         SD=sd(value))%>%
  ungroup()

Minolta_sum<- Minolta%>%
  dplyr::select(Treatment, Position, variable, Mean, SD)%>%
  unique()

  #Figure
  ggplot(data = Minolta,aes(x=Treatment, y=value, fill=Treatment))+ 
    #stat_summary(fun.data = mean_sdl, geom = "errorbar")+ 
    #stat_summary(fun = mean, geom = "bar") +
    geom_violin()+
    #geom_errorbar(aes(ymax=Mean+SD,ymin=Mean-SD),width=0.4)+
    #geom_bar(aes(x=Treatment, y=Mean),stat ="identity",width = 0.8)+
    #geom_point(position = position_jitter(width = 0.2),color="#3182bd",size=1.1)+
    theme_bw()+
    scale_x_discrete(labels=c("Con","Chole","Phyto","DHA"))+
    facet_grid(variable~Position,scales = "free")+
  scale_fill_manual(values=c("#bababa","#fdae61","#b8e186","#a6cee3"))
  
  Minolta_aov <- split(Minolta,list(Minolta$variable,Minolta$Position))
  aov_result<- list()
  posthoc_result <- list()
  ResultAnova <- list()
  for (i in 1:length(Minolta_aov)){
  aov_result[[i]]<- aov(value~Treatment,Minolta_aov[[i]])
  ResultAnova[[i]]= as.data.frame(summary(aov_result[[i]])[[1]])[1,5,drop=F]%>%
    mutate(variable=names(Minolta_aov)[[i]])
  posthoc_result[[i]]<- DunnettTest(value~Treatment,data = Minolta_aov[[i]], control="Control")[[1]][,"pval"]%>%
            t()%>%
            as.data.frame()%>%
            mutate(variable=names(Minolta_aov)[[i]])
 }
  SumAnova <- do.call(rbind.data.frame,ResultAnova)
  SumMulti<- do.call(rbind.data.frame,posthoc_result)
    
    mutate(Group_1 = gsub("\\-.*","",Group),
           Treatment_1 = gsub("\\:.*","",Group_1),
           Position_1 = gsub(".*\\:","",Group_1),
           Group_2 = gsub(".*\\-","",Group),
           Treatment_2 = gsub("\\:.*","",Group_2),
           Position_2 = gsub(".*\\:","",Group_2))%>%
    filter(Position_1==Position_2)

  Minolta_sum <- Minolta%>%
    group_by(Position,variable,Treatment)%>%
    mutate(Mean=mean(value),
           SD=sd(value),
           value=paste(round(Mean, digits=1),"±",round(SD, digits=1)))%>%
    ungroup()%>%
    dplyr::select(Position,variable,Treatment,value)%>%
    unique()%>%
    dcast(Treatment~Position+variable)
  
  
```


## RNA-seq data analysis

```{r,include=F,fig.height=12}
# read count table

Data <- read.csv(".raw/Raw_count.txt", sep="")
Data <- Data %>%
  rownames_to_column("rowname")%>%
  dplyr::mutate(rowname=gsub(".*cds_","",rowname),
                rowname=gsub("P_","P-",rowname),
                rowname=gsub("_.*","",rowname),
                rowname=gsub("P-","P_",rowname))%>%
  column_to_rownames("rowname")
meta <- read_excel(".raw/720020 AstUP results_NMBU_NTNU.xlsx")%>%
  mutate(Fish=as.character(Fish))


TPM_Data <- read.csv(".raw/TPM.txt",sep = "")
TPM_Data <- TPM_Data %>%
  rownames_to_column("rowname")%>%
  dplyr::mutate(rowname=gsub(".*cds_","",rowname),
                rowname=gsub("P_","P-",rowname),
                rowname=gsub("_.*","",rowname),
                rowname=gsub("P-","P_",rowname))%>%
  column_to_rownames("rowname")%>%
  as.data.frame()


# make new meta data
sample_meta <- colnames(Data)%>%
  as.data.frame()%>%
  mutate(name = gsub(".*\\c","",.),
         name = gsub("_S.*","",name),
         Fish = gsub("\\..*","",name)%>%
           as.character(),
         Tank = gsub(".*\\.","Tank",name))%>%
  dplyr::select(-name)%>%
  left_join(meta,by=c("Fish","Tank"))%>%
  mutate(group= paste(Tank,Fish,Treatment, sep = "_"),
         Treatment= factor(Treatment, levels = c("Control","Cholesterol","Phytosterol","DHA")))%>%
  column_to_rownames(".")
sample_meta$sample <- rownames(sample_meta)
  
cpms = cpm(Data,log = T)
keep= rowSums(cpms>1)>=8 
Data = Data[keep,]

d=DGEList(counts=Data,group = sample_meta$group)
d=calcNormFactors(d)

plotMDS(d,labels = sample_meta$Tank)

# PCA
ntop <- 500
rv <- apply(cpm(d, log = TRUE, prior.count = 2), 1, var)
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))]
mat <- t( cpm(d, log = TRUE, prior.count = 2)[select, ] )
pca_all <- prcomp( mat, center = TRUE )

pca <-prcomp(mat, center = TRUE )$x[,1:6] %>% 
  as.data.frame()%>% 
  rownames_to_column('sample') %>%
  dplyr::left_join(sample_meta,by="sample")%>%
  dplyr::mutate(percPC1 = (pca_all$sdev[1]^2/sum(pca_all$sdev^2)),
                percPC2 = (pca_all$sdev[2]^2/sum(pca_all$sdev^2)),
                label=paste(Tank,Fish,sep = "_"))

ggplot(pca, aes(PC1, PC2)) + 
  geom_hline(yintercept=0, colour="gray75") + 
  geom_vline(xintercept=0, colour="gray75") + 
  geom_point(aes(fill=Treatment),shape=21,size=4) + 
  coord_equal()+
  scale_fill_manual(values=c("#bababa","#fdae61","#b8e186","#a6cee3"))+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"))+
  xlab("PC1 (15.7%)")+ylab("PC2 (8.7%)")+
  geom_text_repel(aes(label=label))

PC1_gene<- data.frame(sort(abs(pca$rotation[,"PC1"]),decreasing=TRUE)[1:50])%>%
  rownames_to_column("caption")%>%
  mutate(caption=gsub("\\..*","",caption))

recs <- entrez_summary(db = "protein", id = c(DEA_DHA_sig$gene_id))


Gene_TB <- list()
for (i in 1:length(recs)){
Gene_TB[[i]]<- extract_from_esummary(recs[[i]], c("uid","caption","title"))%>%
  as.data.frame()
}
Gene_TB<- do.call(rbind.data.frame,Gene_TB)
#all_links_sep  <- entrez_link(db="gene", dbfrom="protein", id=Gene_TB$uid, by_id=TRUE)
#Protein_Gene_ID<- lapply(all_links_sep, function(x) #x$links$protein_gene)
#Protein_Gene_ID<- do.call(rbind.data.frame,Protein_Gene_ID)%>%
 # cbind(Gene_TB)
#colnames(Protein_Gene_ID)<- c("Gene_ID","Protein_ID","AA_ID","name")

#Cho_ID_name <- read.csv("Cho_sig_IDvsname.csv")

Design<- model.matrix(~Treatment,data = sample_meta)
rownames(Design)<-d$sample$group

d<-estimateDisp(d,Design)

d.glm<-glmFit(d,Design)

#DEA 
## Cholesterol


DEA_Chol<- glmLRT(d.glm,coef = 2)
DEA_Chol <- as.data.frame(topTags(DEA_Chol,n=Inf))
DEA_Chol_sig <- DEA_Chol[DEA_Chol$FDR<0.05&abs(DEA_Chol$logFC)>1,]%>%
  rownames_to_column("gene_id")%>%
  mutate(caption=gsub("\\..*","",gene_id))%>%
  left_join(Gene_TB,by="caption")%>%
  mutate(name=gsub(" \\[On.*","",title),
         name=gsub("isoform.*","",name))

Cho_TPM_sig <- cpm(d)%>%
  as.data.frame()%>%
  filter(rownames(.)%in%DEA_Chol_sig$gene_id)%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("sample")%>%
  merge(sample_meta%>%
              dplyr::select(group,sample),by="sample")%>%
  column_to_rownames("group")%>%
  dplyr::select(-sample)%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("gene_id")%>%
  merge(DEA_Chol_sig%>%
          dplyr::select(gene_id,name),by="gene_id")%>%
  dplyr::select(contains("Con"),contains("Cho"),name)

DEA_Phyto<- glmLRT(d.glm,coef = 3)
DEA_Phyto <- as.data.frame(topTags(DEA_Phyto,n=Inf))
DEA_Phyto_sig <- DEA_Phyto[DEA_Phyto$FDR<0.05&abs(DEA_Phyto$logFC)>1,]%>%
  rownames_to_column("gene_id")%>%
  mutate(caption=gsub("\\..*","",gene_id))%>%
  left_join(Gene_TB,by="caption")%>%
  mutate(name=" [.*")%>%
  mutate(name=gsub(" \\[On.*","",title),
         name=gsub("isoform.*","",name))%>%
  unique()

PS_TPM_sig <- cpm(d)%>%
  as.data.frame()%>%
  filter(rownames(.)%in%DEA_Phyto_sig$gene_id)%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("sample")%>%
  merge(sample_meta%>%
              dplyr::select(group,sample),by="sample")%>%
  column_to_rownames("group")%>%
  dplyr::select(-sample)%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("gene_id")%>%
  merge(DEA_Phyto_sig%>%
          dplyr::select(gene_id,name),by="gene_id")%>%
  dplyr::select(contains("Con"),contains("Phy"),name)%>%
  unique()

DEA_DHA<- glmLRT(d.glm,coef = 4)
DEA_DHA <- as.data.frame(topTags(DEA_DHA,n=Inf))
DEA_DHA_sig <- DEA_DHA[DEA_DHA$FDR<0.05&abs(DEA_DHA$logFC)>1,]%>%
  rownames_to_column("gene_id")%>%
  mutate(caption=gsub("\\..*","",gene_id))%>%
  left_join(Gene_TB,by="caption")%>%
  mutate(name=" [.*")%>%
  mutate(name=gsub(" \\[On.*","",title),
         name=gsub("isoform.*","",name))


DHA_TPM_sig <- cpm(d)%>%
  as.data.frame()%>%
  filter(rownames(.)%in%DEA_DHA_sig$gene_id)%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("sample")%>%
  merge(sample_meta%>%
              dplyr::select(group,sample),by="sample")%>%
  column_to_rownames("group")%>%
  dplyr::select(-sample)%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("gene_id")%>%
  merge(DEA_DHA_sig%>%
          dplyr::select(gene_id,name),by="gene_id")%>%
  dplyr::select(contains("Con"),contains("DHA"),name)%>%
  unique()


```


```{r,fig.height=8,fig.width=8}

pheatmap(Cho_TPM_sig[1:12],labels_row = Cho_TPM_sig$name,cluster_cols = F,scale = "row", color = colorRampPalette(c("navy", "white", "firebrick3"))(50),show_colnames = F)


pheatmap(PS_TPM_sig[1:12],labels_row = PS_TPM_sig$name,cluster_cols = F,scale = "row", color = colorRampPalette(c("navy", "white", "firebrick3"))(50),show_colnames = F)


pheatmap(DHA_TPM_sig[1:12],labels_row = DHA_TPM_sig$name,cluster_cols = F,scale = "row", color = colorRampPalette(c("navy", "white", "firebrick3"))(50),show_colnames = F)

```




```{r}
Chol_sig_ID<- entrez_link(db="gene", dbfrom="protein", id=DEA_Chol_sig$caption, by_id=TRUE)%>%
  lapply(function(x) x$links$protein_gene)%>%
  do.call(rbind.data.frame,.)%>%
  cbind(DEA_Chol_sig$caption)
colnames(Chol_sig_ID) <- c("Gene_id","Protein_ID")
kk <- enrichKEGG(gene         = Chol_sig_ID$Gene_id,
                 organism     = 'omy',
                 pvalueCutoff = 0.5,
                 universe = )
kk_result <- as.data.frame(kk)%>%
  arrange(-pvalue)%>%
  mutate(Description=factor(Description,levels = .$Description))
ggplot(kk_result,aes(x=Description,y=-log10(pvalue)))+
  geom_point(aes(size=Count),color="#fdae61")+
  coord_flip()+
  theme_bw()+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"))

DEA_Phyto<- glmLRT(d.glm,coef = 3)
DEA_Phyto <- as.data.frame(topTags(DEA_Phyto,n=Inf))
DEA_Phyto_sig <- DEA_Phyto[DEA_Phyto$FDR<0.05,]%>%
  rownames_to_column("gene_id")%>%
  mutate(caption=gsub("\\..*","",gene_id))%>%
  left_join(Gene_TB,by="caption")%>%
  mutate(name=" [.*")%>%
  mutate(name=gsub(" \\[On.*","",title),
         name=gsub("isoform.*","",name))

DEA_DHA<- glmLRT(d.glm,coef = 4)
DEA_DHA <- as.data.frame(topTags(DEA_DHA,n=Inf))
DEA_DHA_sig <- DEA_DHA[DEA_DHA$FDR<0.05,]%>%
  rownames_to_column("gene_id")%>%
  mutate(caption=gsub("\\..*","",gene_id))%>%
  left_join(Gene_TB,by="caption")%>%
  mutate(name=" [.*")%>%
  mutate(name=gsub(" \\[On.*","",title),
         name=gsub("isoform.*","",name))

# Add gene ID and name for the protein ID
recs <- entrez_summary(db = "protein", id = c(DEA_Chol_sig$caption,DEA_Phyto_sig$caption,DEA_DHA_sig$caption))
Gene_TB <- list()
for (i in 1:length(recs)){
Gene_TB[[i]]<- extract_from_esummary(recs[[i]], c("uid","caption","title"))%>%
  as.data.frame()
}
Gene_TB <- do.call(rbind.data.frame,Gene_TB)%>%
  unique()

# Sig table
Sig_TB <- rbind(DEA_Chol_sig%>%
                  mutate(group="Cholesterol_sig"),
                DEA_Phyto_sig%>%
                  mutate(group="Phytosterol_sig"),
                DEA_DHA_sig%>%
                  mutate(group="DHA_sig"))

Total_Cho <- DEA_Chol%>%
  rownames_to_column("caption")%>%
  mutate(caption=gsub("\\..*","",caption))%>%
  left_join(Gene_TB,by="caption")

write.csv(Sig_TB,"../Sig_genes_TB.csv")

# Figure total DEG

Sig_TB_sum <- Sig_TB%>%
  mutate(dir = ifelse(logFC >0,"up-regulation","down-regulation"))%>%
  group_by(dir,group)%>%
  mutate(Num = n())%>%
  ungroup()%>%
  dplyr::select(group,dir,Num)%>%
  unique()%>%
  mutate(Num = ifelse(dir=="down-regulation",-Num,Num),
         group = gsub("_sig","",group))%>%
  mutate(group=factor(group,levels=c("Cholesterol","Phytosterol","DHA")))

ggplot(Sig_TB_sum)+
  geom_bar(aes(x=group,y=Num,fill=group),color="black",alpha=0.8,width=0.6,stat = "identity")+
  theme_bw()+
  geom_hline(yintercept = 0)+
  scale_y_continuous(limits = c(-60,60))+
  xlab("Treatment")+
  ylab("Total number of significant genes")+
  scale_fill_manual(values=c("#fdae61","#b8e186","#a6cee3"))+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"))
  
Sig_label <- read.csv("Sig_genes_TB.csv")%>%
  mutate(sig=ifelse(pValue<0.05,"sig","non-sig"))

ggplot(data = DEA_Chol,aes(x=logFC,y=-log10(PValue),color=sig))+
  geom_point(aes(size=sig))+
  scale_size_manual(values = c(1.4,2.5))+
  geom_hline(yintercept = -log10(0.000115),
             linetype = "dashed")+ 
  geom_vline(xintercept = 0,
             linetype = "dashed") + 
  theme_bw()+
  scale_y_continuous(expand = c(0,0),limits = c(0,26))+
  scale_color_manual(values = c("black","#fdae61"))+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"))+
  geom_text_repel(data = Sig_label,aes(x=Log2.fold.change,y=-log10(pValue),label=Pathway),color="black")

```


```{r}
# PCA not good
ntop <- 500
rv <- apply(cpm(Data, log = TRUE, prior.count = 2), 1, var)
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))]
mat <- t( cpm(Data, log = TRUE, prior.count = 2)[select, ] )
pca.all <- prcomp( mat, center = TRUE )

pca <-prcomp( mat, center = TRUE )$x[,1:6] %>% 
  as.data.frame()  %>% 
  rownames_to_column("sample")%>%
  dplyr::left_join(sample_meta,by="sample")%>%
  dplyr::mutate(percPC1 = (pca.all$sdev[1]^2/sum(pca.all$sdev^2)),
                percPC2 = (pca.all$sdev[2]^2/sum(pca.all$sdev^2)),
                Treatment = gsub("Cholesterol","CH",Treatment),
                Treatment = gsub("Phytosterol","PS",Treatment),
                Treatment = gsub("Control","C",Treatment),
                group1 = paste(Tank,Treatment, sep = "_"))

Sig_label <- read.csv("Sig_genes_TB.csv")

p_PCA <- ggplot(pca, aes(PC1, PC2, color=Treatment,group=group, label=Treatment)) + 
  geom_hline(yintercept=0, colour="gray75") + 
  geom_vline(xintercept=0, colour="gray75") + 
  geom_text(size=5) + coord_equal()+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"),
        legend.position = "None")+
  xlab("PC1 (13.6%)")+
  ylab("PC2 (7.6%)")+
  ggtitle("PCA transcriptome")+
  geom_text_repel()


```


```{r}
# TPM of individual genes
TPM_TB <- melt(TPM_Data%>%
  rownames_to_column("Transcript_ID"))%>%
  as.data.frame()%>%
  mutate(name = gsub(".*\\c","",variable),
         name = gsub("_S.*","",name),
         Fish = gsub("\\..*","",name),
         Tank = gsub(".*\\.","Tank",name))%>%
  dplyr::select(-name)%>%
  left_join(meta%>%
              dplyr::select(Tank,Fish,Treatment,Plasma_Asta),by=c("Tank","Fish"))%>%
  mutate(group= paste(Tank,Fish,Treatment, sep = "_"),
         Treatment = factor(Treatment, levels = c("Control","Cholesterol","Phytosterol","DHA")))%>%
  arrange(Transcript_ID,Treatment)

ggplot(TPM_TB%>%
         filter(Transcript_ID=="XP_036805169.1"),
       aes(x=Treatment,y=value,fill=Treatment))+
  geom_boxplot()+
  geom_point(shape=21)+
  theme_bw()+
  ggtitle("Acetyl CoA synthetase")


# ABC transporters
#abcg8  XP_021432037.2
#abcg2  XP_036828769.1,XP_036828768.1,XP_036828767.1,XP_036828766.1
#abcg5  XP_021432041.1
#abcg1 & abcg4 not expresse
#abcg1a XP_036842961.1, XP_036810103.1,XP_036789596.1
#fasn 

Trans_list <- data.frame(
  genename = c("abcg8","acbg2b_1","abcg2b_2","acbg2b_3","abcg2b_4","acbg5","acbg2a_1","abcg2a_2","abcg2a_3","fasn"),
  Transcript_ID =  c("XP_021432037.2","XP_036828769.1","XP_036828766.1","XP_036828767.1","XP_036828768.1","XP_021432041.1","XP_036842961.1",
               "XP_036810103.1","XP_036789596.1")
)

Transporter_TPM <-left_join(TPM_TB,Trans_list,by="Transcript_ID")%>%
  filter(!is.na(genename))%>%
  group_by(genename,Treatment)%>%
  mutate(Mean = mean(value),
         sd = sd(value))%>%
  ungroup()

ggplot(Transporter_TPM)+
  geom_boxplot(aes(x=Treatment,y=value,fill=Treatment),alpha=0.8,outlier.shape = NA)+
  facet_wrap(~genename,scales = "free")+
  geom_point(aes(x=Treatment,y=value,fill=Treatment),shape=21,position = position_dodge2(width = 0.4))+
  theme_bw()

#interesting genes
Gene_TB <- TPM_TB%>%
  mutate(caption = gsub("\\..*","",Transcript_ID))%>%
  left_join(Sig_TB,by="caption")%>%
  filter(!is.na(logFC))
  
ggplot(Gene_TB%>%
         filter(uid =="1925143124"|
                  uid =="1211303628"|
                  uid == "1925176613"|
                  uid == "1211243549"|
                  uid == "185134415"|
                  uid =="1925099030"|
                  uid =="1925121968")%>%
        mutate(title = gsub(" \\[.*","",title),
               title = paste(title,Transcript_ID,sep="_")))+
  geom_boxplot(aes(x=Treatment,y=value,fill=Treatment),alpha=0.8,outlier.shape = NA)+
  facet_wrap(~title,scales = "free")+
  geom_point(aes(x=Treatment,y=value,fill=Treatment),shape=21,position = position_dodge2(width = 0.4))+
  theme_bw()

# sterol regulatory element binding protein
# srebp1 XP_036797488.1 XP_036797487.1 ALM01569.1 XP_036794890.1 XP_036794889.1 XP_036801651.1 XP_036801650.1 XP_036801649.1 XP_021449513.1 XP_021449512.1 XP_021480773.2 XP_021480772.2 XP_021480770.2 XP_021413728.2 XP_021413727.2 XP_021413726.2

Trans_list <- data.frame(
  genename = c("srebp1_1","srebp1_2","srebp1_3","srebp1_4","srebp1_5","srebp2_1","srebp2_2","srebp2_3","srebp2_4","srebp2_5","srebp2_6"),
  Transcript_ID =  c("XP_036797488.1","XP_036797487.1","ALM01569.1","XP_036794890.1","XP_036794889.1","XP_021480773.2","XP_021480772.2","XP_021480770.2","XP_021413728.2","XP_021413727.2","XP_021413726.2")
)

SREBP_TPM <-left_join(TPM_TB,Trans_list,by="Transcript_ID")%>%
  filter(!is.na(genename))%>%
  group_by(genename,Treatment)%>%
  mutate(Mean = mean(value),
         sd = sd(value))%>%
  ungroup()

ggplot(SREBP_TPM)+
  geom_boxplot(aes(x=Treatment,y=value,fill=Treatment),alpha=0.8,outlier.shape = NA)+
  facet_wrap(~genename,scales = "free")+
  geom_point(aes(x=Treatment,y=value,fill=Treatment),shape=21,position = position_dodge2(width = 0.4))+
  theme_bw()


```


## Continuous variable analysis

Instead of calculating differential expression between the samples classified by discrete phenotypic traits, e.g. cholesteorl diet versus control, this analysis will identify correlation between gene expression and continuous traits.

The measurement of plasma astaxathin level could be an continuous variable. 

Using edgeR (version `r packageDescription("edgeR")$Version`) the continuous measurements can be used as a predictor variable.

### Running edgeR

Genes with low expression were filtererd by demanding that at least 8 samples must have a count per million (CPM) > 1.

Dispersions were estimated using Diet as blocking factor.

Models use a single predictor variable which has been scaled (i.e. mean = 0 and sd = 1) so that estimated slope parameters are compareable

```{r}
# Amount of plasma astaxanthin level was used as continuous variable
# First check the data

ggplot(sample_meta,aes(x=sample,y=Plasma_Asta,color=Treatment))+
  geom_point()

# convert counts to matrix

meta_new <- sample_meta%>%
  dplyr::select(Treatment,Tank,Fish,Plasma_Asta )%>%
  filter(Treatment =="Cholesterol"|Treatment=="Control")
exprCountMat <- as.matrix(Data[,colnames(Data)%in%rownames(meta_new)])

# 
y <- DGEList(counts=exprCountMat)

# calculate library size normalization factors
y <- calcNormFactors(y)

# filter low expressed genes, here I filter out gene that not expressed in any samples, because 0 value makes it difficult for regression
keepGenes <- rowSums(cpm(y)>1) >= 6
y <- y[keepGenes,keep.lib.sizes=FALSE]

# scale the variables so that they are comparable
meta_new_scaled <-
  meta_new %>%
  mutate_at( vars(Plasma_Asta), scale )

# dispersion
y <- estimateDisp(y, model.matrix(~Plasma_Asta+Treatment,data = meta_new_scaled))


# try different variables as only factor
designs <- list(
  Plasma_Asta = model.matrix(~ Plasma_Asta, data = meta_new_scaled),
  Plasma_Asta_Treatment = model.matrix(~Plasma_Asta+Treatment,data = meta_new_scaled)
)

lapply( designs, function(design){
  glmQLFit(y,design) %>% 
    glmQLFTest(coef=2) %>% 
    with(table) %>% 
    mutate( padj = p.adjust(PValue,"fdr") )
}) -> res  

# ggplot for single gene
Data_sg <- TPM_Data[,colnames(TPM_Data)%in%rownames(meta_new)]%>%
  filter(rownames(.)=="XP_036790190.1")%>%
  t()%>%
  as.data.frame()%>%
  rename("TPM"="XP_036790190.1")%>%
  merge(meta_new_scaled,by="row.names")

ggplot(Data_sg,aes(x=TPM,y=Plasma_Asta))+
  geom_point(aes(color=Treatment))+
  geom_smooth(method = "lm")+
  ggtitle("XP_036799474.1_ABCB7")+
  theme_bw()

# List all sig genes
res_sig <- res$Plasma_Asta%>%
  filter(PValue<0.0005)
recs <- entrez_summary(db = "protein", id = c(rownames(res_sig)))
Gene_TB <- list()
for (i in 1:length(recs)){
Gene_TB[[i]]<- extract_from_esummary(recs[[i]], c("uid","caption","title"))%>%
  as.data.frame()
}
Gene_TB <- do.call(rbind.data.frame,Gene_TB)%>%
  unique()

Data_sg <- TPM_Data[rownames(TPM_Data)%in%rownames(res_sig),]%>%
  t()%>%
  as.data.frame()%>%
  merge(meta_new_scaled,by="row.names")%>%
  melt(id.vars=c("Tank","Fish","Plasma_Asta","Treatment","Row.names"))

ggplot(Data_sg,aes(x=value,y=Plasma_Asta))+
  geom_point(aes(color=Treatment))+
  geom_smooth(method = "lm")+
  facet_wrap(~variable,scales = "free")+
  theme_bw()

# Sig table
Sig <- rbind(res_sig)%>%
  rownames_to_column("caption")%>%
  mutate(caption=gsub("\\..*","",caption))%>%
  left_join(Gene_TB,by="caption")

# fit <- glmQLFit(y, model.matrix( ~ 1 + LCT  + relWeight,data = sampleMeta_NO_scaled))
# qlt <-  glmQLFTest(glmfit = fit,coef = 2:3)
# table(p.adjust(qlt$table$PValue,"fdr")<0.05)
# sapply(res,function(x){table(x$padj<0.05)})
# sampleMeta_NO %>% select(MCT:ACT) %>% summarise_all(function(x){sum(is.na(x))})
```


# New proteomic analysis
# Using package DEP
# The difference is that the NA value is filtered and the data set was normalized
# link for proteomic analysis https://bioconductor.org/packages/devel/bioc/vignettes/DEP/inst/doc/DEP.html

```{r,include=F}
library(readxl)
library(DEP)
library(tidyverse)
library(ggplot2)
library(ggrepel)
library(pheatmap)
library(reshape2)
library(ggfortify)
library(broom)

```


```{r}
#read raw file
library(readr)
proteinGroups <- read_delim("../../Proteomics/raw/proteinGroups.txt", 
                            delim = "\t", escape_double = FALSE, 
                            col_types = cols(`Only identified by site` = col_character(),Reverse=col_character(),`Potential contaminant` =col_character()), 
                            trim_ws = TRUE)
Protein_ID <- read_delim("../../Proteomics/raw/Protein_ID.txt", 
     delim = "\t", escape_double = FALSE, 
     col_types = cols(protein_ID = col_character()), 
     trim_ws = TRUE)
colnames(proteinGroups)
DB_meta <- read_excel("../../../data/RNA-SEQ/meta/720020 AstUP results_NMBU_NTNU.xlsx",sheet = 1)

# First is to annotate uniprot with gene name
# This was done by uniprot website, transfer uniprot to gene name
Uniprot_to_genename <- read.table("../../Proteomics/raw/Unipro_to_GeneName.txt",header = T)
Data <- proteinGroups%>%
  mutate(Uniprot=gsub(";.*","",`Protein IDs`))

Data <- Data%>%
  filter(is.na(Reverse))%>%
  filter(is.na(`Potential contaminant`))

Data<- Data%>%
  merge(Uniprot_to_genename,by="Uniprot")



# remove sample #24 becasue low protein identified
Data<- Data%>%
  dplyr::select(-`LFQ intensity 24_20210308111738`)

colnames_new <- colnames(Data%>%
           dplyr::select(contains("LFQ intensity")))%>%
  as.data.frame()%>%
  mutate(protein_ID=gsub("LFQ intensity ","",.),
         protein_ID=gsub("_.*","",protein_ID))%>%
  rename("old_col"=".")%>%
  left_join(Protein_ID,by="protein_ID")%>%
  separate(sample_ID, into=c("Tank","other","Fish"),sep="_")%>%
  mutate(Tank=paste("Tank",Tank,sep = ""))%>%
  left_join(DB_meta%>%
              dplyr::select(Tank,Treatment),by="Tank")%>%
  mutate(Treatment=factor(Treatment,levels = c("Control","Phytosterol","DHA","Cholesterol")))%>%
  arrange(Treatment)%>%
  unique()%>%
  mutate(replicate=c(1:8,1:9,1:9,1:9),
         new_col = paste(Treatment,replicate,sep = "_"))%>%
  dplyr::select(old_col,new_col)

Data <- Data%>%
  t()%>%
  as.data.frame()%>%
  rownames_to_column("old_col")%>%
  left_join(colnames_new,by="old_col")%>%
  mutate(new_col=ifelse(is.na(new_col),old_col,new_col))%>%
  column_to_rownames("new_col")%>%
  dplyr::select(-old_col)%>%
  t()%>%
  as.data.frame()
#keep column needed
Data <- Data%>%
  dplyr::select(Uniprot,Genename,`Fasta headers`, contains("Control"),contains("Phytosterol"),contains("DHA"),contains("Cholesterol"))%>%
  dplyr::select(-Cholesterol_9)


# Remove duplication
Data$Genename %>% duplicated() %>% any()
Data %>% group_by(Genename) %>% summarize(frequency = n()) %>% 
  arrange(desc(frequency)) %>% filter(frequency > 1)
data_unique <- make_unique(Data, "Genename", "Uniprot", delim = ";")

# Generate a SummarizedExperiment object
  # Getting all sample column
Diet_columns <- grep("Control|Phytosterol|DHA|Cholesterol",colnames(data_unique))
  
  # generate design matrix
  Design_matrix <- colnames(data_unique[,Diet_columns])%>%
    as.data.frame()%>%
    rename("label"=".")%>%
    mutate(condition = gsub("_.*","",label)%>%
             factor(levels = c("Control","Phytosterol","DHA","Cholesterol")),
           replicate = gsub(".*_","",label)%>%
             as.numeric())%>%
    arrange(condition)

#lets try with only cholesterol
Diet_columns<-which(colnames(data_unique)%in%Design_matrix$label)

#make SummarizedExperiment
data_unique[Diet_columns] <- sapply(data_unique[Diet_columns],as.numeric)
data_unique[data_unique==0] <- NA
data_se <- make_se(data_unique, Diet_columns, Design_matrix)

#Filter on missing values
#The dataset contains proteins which are not quantified in all replicates. Some proteins are even only quantified in a single replicate.
# Plot a barplot of the protein identification overlap between samples
plot_frequency(data_se)

# Filter for proteins that are quantified in at least 3/4 of the samples.
data_filt <- filter_proteins(data_se, "fraction", min = 0.75)



# 2161 protein were identified
# Plot a barplot of the number of identified proteins per samples
plot_numbers(data_filt)

# Plot a barplot of the protein identification overlap between samples
plot_coverage(data_filt)

#The data is background corrected and normalized by variance stabilizing transformation (vsn).
# Normalize the data
data_norm <- normalize_vsn(data_filt)
meanSdPlot(data_norm)
plot_normalization(data_filt, data_norm)

# Plot a heatmap of proteins with missing values
plot_missval(data_norm)

# Plot intensity distributions and cumulative fraction of proteins with and without missing values
plot_detect(data_filt)

# Impute missing data using the k-nearest neighbour approach (for MAR)
knn_imputation <- impute(data_filt, fun = "knn", rowmax = 0.9)

#The effect of data imputation on the distributions can be visualized.
# Plot intensity distributions before and after imputation
plot_imputation(data_norm,knn_imputation)

# Differential enrichment analysis  based on linear models and empherical Bayes statistics
# Test every sample versus control
data_diff <- test_diff(knn_imputation, type = "control", control = "Control")


# Denote significant proteins based on user defined cutoffs
dep <- add_rejections(data_diff,alpha = 0.05)

Count_table<-dep@assays@data@listData[[1]]%>%
  as.data.frame()

# PCA of proteins
ntop <- 200
rv <- apply(cpm(Count_table, log = TRUE, prior.count = 2), 1, var)
select <- order(rv, decreasing = TRUE)[seq_len(min(ntop, length(rv)))]
mat <- t( cpm(Count_table, log = TRUE, prior.count = 2)[select, ] )
pca.all <- prcomp( mat, center = TRUE )

pca <-prcomp( mat, center = TRUE )$x[,1:6] %>% 
  as.data.frame()  %>%
  dplyr::mutate(percPC1 = (pca.all$sdev[1]^2/sum(pca.all$sdev^2)),
                percPC2 = (pca.all$sdev[2]^2/sum(pca.all$sdev^2)),
                Treatment = gsub("_.*","",rownames(.)))

ggplot(pca, aes(PC1, PC2, fill=Treatment,label=Treatment)) + 
  geom_hline(yintercept=0, colour="gray75") + 
  geom_vline(xintercept=0, colour="gray75") + 
  geom_point(shape=21)+
  coord_equal()+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"),
        legend.position = "None")+
  xlab("PC1 (46.6%)")+
  ylab("PC2 (7.0%)")+
  ggtitle("PCA Proteome")

plot_pca(dep, x = 1, y = 2, n = 100, point_size = 4,indicate = c("condition"))

autoplot(prcomp( mat, center = TRUE ), data = pca, colour = 'Treatment', frame = TRUE)

#DEP table
data_results <- get_results(dep)%>%
  left_join(Data%>%
              mutate(ID=Uniprot)%>%
              dplyr::select(ID,`Fasta headers`))%>%
  mutate(Genename=gsub("OS=.*","",`Fasta headers`),
         Genename=gsub(".*SALSA |.*ONCMY ","",Genename)
         )%>%
  mutate(Cholesterol_vs_Control_p.adj=p.adjust(Cholesterol_vs_Control_p.val,method = "BH"),
        Phytosterol_vs_Control_p.adj=p.adjust(Phytosterol_vs_Control_p.val,method = "BH"),
        DHA_vs_Control_p.adj=p.adjust(DHA_vs_Control_p.val,method = "BH"))

#sig set as 0.001
DEP_CH_table <- data_results%>%
  dplyr::select(name,ID,Genename,Cholesterol_vs_Control_p.val,Cholesterol_vs_Control_ratio)%>%
  rename(Log2FC = Cholesterol_vs_Control_ratio,
         pvalue = Cholesterol_vs_Control_p.val)%>%
  filter(pvalue<0.005)%>%
  mutate(group="CH")

DEP_PS_table <- data_results%>%
  dplyr::select(name,ID,Genename,Phytosterol_vs_Control_p.val,Phytosterol_vs_Control_ratio)%>%
  rename(Log2FC = Phytosterol_vs_Control_ratio,
         pvalue = Phytosterol_vs_Control_p.val)%>%
  filter(pvalue<0.005)%>%
  mutate(group="PS")

DEP_DHA_table <- data_results%>%
  dplyr::select(name,ID,Genename,DHA_vs_Control_p.val,DHA_vs_Control_ratio)%>%
  rename(Log2FC = DHA_vs_Control_ratio,
         pvalue = DHA_vs_Control_p.val)%>%
  filter(pvalue<0.005)%>%
  mutate(group="DHA")

DEP_TB <- rbind(DEP_CH_table,DEP_PS_table,DEP_DHA_table)

# Figure total DEP
Sig_TB_sum <- DEP_TB%>%
  mutate(dir = ifelse(Log2FC >0,"up-regulation","down-regulation"))%>%
  group_by(dir,group)%>%
  mutate(Num = length(ID))%>%
  ungroup()%>%
  dplyr::select(group,dir,Num)%>%
  unique()%>%
  mutate(Num = ifelse(dir=="down-regulation",-Num,Num))%>%
  mutate(group=factor(group,levels=c("CH","PS","DHA")))

ggplot(Sig_TB_sum)+
  geom_bar(aes(x=group,y=Num,fill=group),color="black",alpha=0.8,width=0.6,stat = "identity")+
  theme_bw()+
  geom_hline(yintercept = 0)+
  scale_y_continuous(limits = c(-20,20))+
  scale_x_discrete(limits=c("CH","PS","DHA"))+
  xlab("Treatment")+
  ylab("Total number of significant PROTEINS")+
  scale_fill_manual(values=c("#fdae61","#b8e186","#a6cee3"))+
  theme(panel.background =  element_rect(fill = "white",colour = "black"),
        panel.border = element_rect(linetype = "solid", fill = NA),
        strip.background = element_rect(fill = "grey",colour = "black"),
        text = element_text(size=16, face = "italic"),
        axis.line = element_line(colour = "black"),
        strip.text = element_text(size=16,face = "italic"),
        axis.text =element_text(color = "black", size=16, face = "plain"),
        axis.text.y = element_text(face = "italic" ),
        plot.title = element_text(size=16,face="italic"))

# heatmap DEP
Count_table <- dep@assays@data@listData[[1]]%>%
  as.data.frame()%>%
  rownames_to_column("name")%>%
  filter(name%in%DEP_CH_table$name)%>%
  left_join(DEP_CH_table%>%
              dplyr::select(name,Genename),by="name")%>%
  dplyr::select(-name)%>%
  dplyr::select(contains("Control"),contains("Cholesterol"),Genename)
pheatmap(Count_table[,1:16],cluster_cols = F,scale = "row",labels_row = Count_table$Genename,color=colorRampPalette(c("#018571", "#f5f5f5", "#a6611a"))(50))

Count_table_PS <- dep@assays@data@listData[[1]]%>%
  as.data.frame()%>%
  rownames_to_column("name")%>%
  filter(name%in%DEP_PS_table$name)%>%
  left_join(DEP_PS_table%>%
              dplyr::select(name,Genename),by="name")%>%
  dplyr::select(-name)%>%
  dplyr::select(contains("Control"),contains("Phytosterol"),Genename)
pheatmap(Count_table_PS[,1:16],cluster_cols = F,scale = "row",labels_row = Count_table$Genename,color=colorRampPalette(c("#018571", "#f5f5f5", "#a6611a"))(50))

Count_table_DHA <- dep@assays@data@listData[[1]]%>%
  as.data.frame()%>%
  rownames_to_column("name")%>%
  filter(name%in%DEP_DHA_table$name)%>%
  left_join(DEP_DHA_table%>%
              dplyr::select(name,Genename),by="name")%>%
  dplyr::select(-name)%>%
  dplyr::select(contains("Control"),contains("DHA"),Genename)
pheatmap(Count_table_DHA[,1:16],cluster_cols = F,scale = "row",labels_row = Count_table$Genename,color=colorRampPalette(c("#018571", "#f5f5f5", "#a6611a"))(50))

```
